R/provider-huggingface.r

```diff
@@ -0,0 +1,58 @@
+#' Chat with a model hosted on Hugging Face Serverless Inference API
+#'
+#' @description
+#' [Hugging Face](https://huggingface.co/) hosts a variety of open-source
+#' and proprietary AI models available via their Inference API.
+#' To use the Hugging Face API, you must have an Access Token, which you can obtain
+#' from your [Hugging Face account](https://huggingface.co/settings/tokens).
+#'
+#' This function is a lightweight wrapper around [chat_openai()], with
+#' the defaults adjusted for Hugging Face. Model defaults to `meta-llama/Llama-3.1-8B-Instruct`.
+#'
+#' ## Known limitations
+#'
+#' * Some models do not support the chat interface or parts of it, for example
+#' `google/gemma-2-2b-it` does not support a system prompt. You will need to
+#'  carefully choose the model.
+#'
+#' @family chatbots
+#' @param api_key The API key to use for authentication. You should not
```

Comment:
Can you please mimic the existing structure from the other chat functions? Or if you think it's worth being more explicit, it'd be useful to create a `api_key_doc(key_name)` function that could generate the docs for all of them, eliminating any inconsistencies or duplication. Please let me know if you need some more hints on the appropriate way to do that in roxygen2.